mlpp_lib/datasets.py

```diff
@@ -596,6 +596,7 @@ def __init__(
         self.shuffle = shuffle
         self.block_size = block_size
         self.num_samples = len(self.dataset.x)
+        # self.num_batches = int(np.ceil(self.num_samples / batch_size))
```

Comment:
what is this about?

Reply:
With the current implementation we never see the whole validation dataset if its length is not a multiple of the batch size.
We could change for this method to ensure all the dataset is seen

Reply:
but this is something unrelated to this PR, isn't it? I suggest to move it to another PR and discuss it there